src/Core/Usecase/Message/ReceiveMessage.php

```diff
@@ -23,36 +23,18 @@
 use Ushahidi\Core\Tool\Validator;
 use Ushahidi\Core\Usecase\CreateUsecase;
 use Ushahidi\Core\Usecase\CreateRepository;
+use Ushahidi\Core\Traits\Events\DispatchesEvents;
 
 use Ushahidi\Core\Exception\ValidatorException;
 
 class ReceiveMessage extends CreateUsecase
```

Comment:
So this is a pretty straightforward usecase now. It looks for a contact, creates one if it didn't exists, then saves the message.